### PR TITLE
Add spec for mentors search

### PR DESCRIPTION
**Pull request by jgradim**

### CODE REVIEW

File: spec/requests/mentors_controller_spec.rb [hunk 1]
```diff
@@ -17,9 +17,40 @@
...
+        expect(json.map{ |u| u[:id] }).to match_array [@mentor2.id, @mentor3.id]
+      end
+
+      it "by multiple params" do
+        get "/mentors", headers: auth_headers, params: { string: "word", career_ids: @career_ids[0..1] }
+        expect(json.map{ |u| u[:id] }).to match_array [@mentor3.id]
+
+        get "/mentors", headers: auth_headers, params: { career_ids: @career_ids[0...1], trait_ids: @trait_ids }
+        expect(json.map{ |u| u[:id] }).to match_array [@mentor1.id]
+      end
```
Comment (jgradim) [inline spec/requests/mentors_controller_spec.rb]: Move search specific testing to user model, keep simple example here

**Comment by jgradim** [comment]
@hugopeixoto updated!

File: spec/requests/mentors_controller_spec.rb [hunk 2]
```diff
@@ -16,10 +16,24 @@
       expect(response.status).to eq 200
     end
 
-    context "filters mentors" do
-      it "by careers"
-      it "by traits"
-      it "by search"
+    it "GET index with filters" do
+      # spied = spy(User)
```
Comment (hugopeixoto) [inline spec/requests/mentors_controller_spec.rb]: Wat

File: spec/models/user_spec.rb [hunk 3]
```diff
@@ -24,4 +24,76 @@
...
+      mentor3 = create(:user, :mentor, name: "three word", traits: @traits)
+      mentor4 = create(:user, :mentor, location: "the word")
+
+      mentors = User.search(string: "word")
+      expect(mentors).to match_array [mentor1, mentor2, mentor3, mentor4]
+
+      mentors = User.search(string: "one three")
+      expect(mentors).to match_array []
+
+      mentors = User.search(string: "trait") # all generated traits have description = "trait#{index}"
```
Comment (hugopeixoto) [inline spec/models/user_spec.rb]: replace `"trait"` and its comment with `@traits.first.description`

File: spec/models/user_spec.rb [hunk 4]
```diff
@@ -24,4 +24,76 @@
...
   end
+
+  context "searches and filters mentors" do
+    before do
+      @careers = 3.times.map{ create(:career) }
+      @traits = 3.times.map{ create(:trait) }
+    end
+
+    it "by careers" do
+      mentor1 = create(:user, :mentor, careers: @careers[0..1])
```
Comment (hugopeixoto) [inline spec/models/user_spec.rb]: use `@careers.first(2)`

File: spec/models/user_spec.rb [hunk 5]
```diff
@@ -24,4 +24,76 @@
...
+      mentor2 = create(:user, :mentor)
+      mentor3 = create(:user, :mentor, careers: @careers[1..-1])
```
Comment (hugopeixoto) [inline spec/models/user_spec.rb]: user `@careers.last(2)`

File: spec/models/user_spec.rb [hunk 6]
```diff
@@ -24,4 +24,76 @@
...
+
+      mentors = User.search(career_ids: [ @careers[1].id ])
+      expect(mentors).to match_array [mentor1, mentor3]
+
+      mentors = User.search(career_ids: [ @careers[-1].id ])
```
Comment (hugopeixoto) [inline spec/models/user_spec.rb]: `@careers.last.id`

File: spec/models/user_spec.rb [hunk 7]
```diff
@@ -24,4 +24,76 @@
...
+      expect(mentors).to match_array [mentor3]
+    end
+
+    it "by traits" do
```
Comment (hugopeixoto) [inline spec/models/user_spec.rb]: Same as above

**Review by hugopeixoto — CHANGES_REQUESTED** [review]

**Pull request merged**